### ISSUE TITLE
API: disallow div/floordiv/pow operators for BooleanArray ?

### ISSUE THREAD

**Issue by jorisvandenbossche** [pandas-dev/pandas#41165]
Currently, for the plain `bool` dtype we explicitly check for some operations and raise an error, while those actually work in numpy. For example:

```
In [5]: arr = np.array([True, False, True])

In [6]: arr / True
Out[6]: array([1., 0., 1.])

In [7]: pd.Series(arr) / True
...
NotImplementedError: operator '/' not implemented for bool dtypes
```

This is done for the division and power operations (`not_allowed=frozenset(("/", "//", "**"):

https://github.com/pandas-dev/pandas/blob/934cad6ab61b867c6ae54941c5cd87340d44b80a/pandas/core/computation/expressions.py#L215-L218

**Comment by dsaxton** [comment]
@jorisvandenbossche What's the argument for disallowing at the moment? To me it seems more natural / Pythonic to allow this operation since booleans are essentially ints.

**Comment by jorisvandenbossche** [comment]
I am not sure what the historic reasons are. 
Maybe because those operations were regarded as not that useful (although I would say it's up to the user to decide that), or as potentially confusing because they don't have a "boolean" interpretation, but only a numerical one (eg `+` and `*` still are a boolean operation resulting in booleans, and eg `-` raises an error in numpy about not being supported for booleans). 
It's only for division and power that the booleans are interpreted as numeric values (and currently in pandas the user needs to be explicit about such casting).

**Comment by dsaxton** [comment]
@jorisvandenbossche Yeah, I can see the argument of "why would anyone do this?" but if it's easy to allow and makes for greater consistency with `numpy` and Python more generally, I personally like the idea of allowing this rather than making a special case here.

**Comment by jbrockmendel** [comment]
conditional on the Series behavior (which i wouldnt object to deprecating), I lean towards having BooleanArray behave like Series, i.e. raising here

**Comment by jbrockmendel** [comment]
Current thought here: BooleanArray (and IntegerArray and FloatingArray) ops should be wrappers around their core.ops.array_ops counterparts.  This will allow for pushing more logic down from BooleanArray/NumericArray into BaseMaskedArray, which in turn will make it easier to extend BaseMaskedArray to wrap arbitrary dtypes.

**Comment by wence-** [comment]
> or as potentially confusing because they don't have a "boolean" interpretation

FWIW, booleans are a canonical identification for GF(2) so I would argue that these operations _are_ all well-defined and have a unique interpretation.

**Comment by jbrockmendel** [comment]
>  GF(2)

We don't have an official policy on this, but in general pandas is more averse to overflows than numpy, which corresponds to not treating arithmetic as modular.
